assets/js/googlesitekit/site/connection.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import { get } from 'googlesitekit-api';
+
+// Actions
+const FETCH_CONNECTION = 'FETCH_CONNECTION';
+const RECEIVE_CONNECTION = 'RECEIVE_CONNECTION';
+const RECEIVE_CONNECTION_FAILED = 'RECEIVE_CONNECTION_FAILED';
+
+export const actions = {
+	fetchConnection() {
+		return {
+			payload: {},
+			type: FETCH_CONNECTION,
+		};
+	},
+
+	receiveConnection( connection ) {
+		invariant( connection, 'connection is required.' );
+
+		return {
+			payload: { connection },
+			type: RECEIVE_CONNECTION,
+		};
+	},
+
+	receiveConnectionFailed( error ) {
+		invariant( error, 'error is required.' );
+
+		return {
+			payload: { error },
+			type: RECEIVE_CONNECTION_FAILED,
+		};
+	},
+};
+
+export const controls = {
+	[ FETCH_CONNECTION ]: async () => {
+		return get( 'core', 'site', 'connection' );
+	},
+};
+
+export const reducer = ( state, action ) => {
+	switch ( action.type ) {
+		case FETCH_CONNECTION: {
+			return {
+				...state,
+				isFetchingConnection: true,
+			};
+		}
+
+		case RECEIVE_CONNECTION: {
+			const { connection } = action.payload;
+
+			return {
+				...state,
+				isFetchingConnection: false,
+				connection,
+			};
+		}
+
+		case RECEIVE_CONNECTION_FAILED: {
+			const { error } = action.payload;
+
+			return {
+				...state,
+				isFetchingConnection: false,
+				connection: { error, hasError: true },
+			};
+		}
+
+		default: {
+			return { ...state };
+		}
+	}
+};
+
+export const resolvers = {
+	*getConnection() {
+		try {
+			const connection = yield actions.fetchConnection();
+			return actions.receiveConnection( connection );
+		} catch ( err ) {
+			// TODO: Implement an error handler store or some kind of centralized
+			// place for error dispatch...
+			return actions.receiveConnectionFailed( err );
+		}
+	},
+};
+
+export const selectors = {
+	getConnection: ( /* state */ ) => {
+		return 'should fail';
```

Comment:
Something is wrong here; this is never being called despite the tests passing and the selector returning this information. I'm a bit perplexed, so I think something obvious is eluding me here.

Code coverage caught this error by the way, so it's already paying off 😅 

<img width="435" alt="Screenshot 2020-02-28 at 20 40 55" src="https://user-images.githubusercontent.com/90871/75585711-9faa6900-5a6a-11ea-801e-a2c2f3fe090e.png">

Reply:
I figured out the problem here, and it's not great. Selectors will override each other, so it looks like my original refactor of `index.js` to become `connection.js` and `reset.js` left the `getConnection()` selector in both files. This meant that `reset.js`'s `getConnection` was being used, which was tough to spot.

I've sorted it, but I guess that's going to be an issue we face when combining multiple sets of reducers/selectors. In order to combat that I'm adding something that errors on duplicate keys for actions/selectors/etc. within the same store, or someone else will run into this and be spinning their wheels like I just did.